### PR TITLE
Integration test up to cast vote

### PR DESCRIPTION
**Pull request by nolanjacobson**

### CODE REVIEW

File: tip-router-operator-cli/tests/integration_tests.rs [hunk 1]
```diff
@@ -202,6 +206,78 @@ impl TestContext {
...
+    assert!(
+        meta_merkle_tree.merkle_root != [0u8; 32],
+        "Meta merkle tree has zero root"
+    );
+
+    Ok(())
+}
+
+// This function would be defined in your main code, not in tests
+pub fn get_merkle_root(
```
Comment (Taylor123) [inline tip-router-operator-cli/tests/integration_tests.rs]: this should not be in the test, it's biz logic

File: tip-router-operator-cli/tests/integration_tests.rs [hunk 2]
```diff
@@ -202,6 +206,78 @@ impl TestContext {
...
+            bank_hash: stake_meta_collection.hash().to_string(),
+            slot: stake_meta_collection.slot(),
+            tip_distribution_program_id: *tip_distribution_program_id,
+        },
+        protocol_fee_bps,
+    )?;
+
+    // Convert to MetaMerkleTree
+    let meta_merkle_tree =
+        MetaMerkleTree::new_from_generated_merkle_tree_collection(merkle_tree_coll)?;
```
Comment (Taylor123) [inline tip-router-operator-cli/tests/integration_tests.rs]: make it have no implementation to start

Comment (nolanjacobson) [reply tip-router-operator-cli/tests/integration_tests.rs]: Okay... addressed.

File: tip-router-operator-cli/tests/integration_tests.rs (hunk 1 above)
Comment (nolanjacobson) [reply tip-router-operator-cli/tests/integration_tests.rs]: Removed

File: tip-router-operator-cli/tests/integration_tests.rs [hunk 3]
```diff
@@ -202,6 +206,42 @@ impl TestContext {
     }
 }
 
+#[tokio::test]
+#[should_panic(expected = "get_merkle_root not implemented")]
```
Comment (tomjohn1028) [inline tip-router-operator-cli/tests/integration_tests.rs]: remove this, we want the test to fail one way or another

File: Cargo.lock [hunk 4]
```diff
@@ -8981,11 +8981,13 @@ version = "0.1.0"
 dependencies = [
  "anchor-lang 0.30.1",
  "anyhow",
+ "base64 0.13.1",
```
Comment (Taylor123) [inline Cargo.lock:8984]: the new deps should be dev-dependencies, should they not?

File: tip-router-operator-cli/Cargo.toml [hunk 5]
```diff
@@ -7,12 +7,14 @@ description = "CLI for Jito Tip Router"
 [dependencies]
 anchor-lang = { workspace = true }
 anyhow = { workspace = true }
+base64 = "0.13"
```
Comment (Taylor123) [inline tip-router-operator-cli/Cargo.toml:10]: the new deps should be dev-dependencies, should they not?

Comment (nolanjacobson) [reply tip-router-operator-cli/Cargo.toml:10]: you are right. resolved.

File: tip-router-operator-cli/scripts/setup-test-ledger.sh [hunk 6]
```diff
@@ -20,20 +21,53 @@ create_keypair () {
...
+    # Create the vote account
+    solana create-vote-account \
+      "$keys_dir/vote_$number.json" \
+      "$keys_dir/identity_$number.json" \
+      "$keys_dir/withdrawer_$number.json" \
+      --commission 1
+  done
+}
+
+
```
Comment (Taylor123) [inline tip-router-operator-cli/scripts/setup-test-ledger.sh:70]: is this still needed with `prepare_keypairs_and_serialize`?

Comment (tomjohn1028) [reply tip-router-operator-cli/scripts/setup-test-ledger.sh:70]: yeah, that's creating the vote accounts

**Pull request merged**